web/src/components/datasets/DatasetDetailPage.tsx

```diff
@@ -142,26 +154,29 @@ const DatasetDetailPage: FunctionComponent<IProps> = props => {
       </Box>
       {tab === 0 && (
         <DatasetInfo
-          datasetFields={dataset.fields}
-          facets={dataset.facets}
-          run={dataset.createdByRun}
+          datasetFields={firstVersion.fields}
+          facets={firstVersion.facets}
+          run={firstVersion.createdByRun}
         />
       )}
       {tab === 1 && <DatasetVersions versions={props.versions} />}
+      {tab === 2 && (props.dataset.name !== undefined || fetchDataset(props.lineageDataset.namespace, props.lineageDataset.name)) && <DatasetColumnLineage columnLineage={props.dataset.columnLineage} />}
```

Comment:
I wanted to achieve the behavior such that backend request Is sent only when a column lineage tab is clicked. 
This is working like that although I think react may have a different approach to achieving that. 

An opinion of a webdev PRO would be welcome!